src/datasources/balances-api/coingecko-api.service.ts

```diff
@@ -399,29 +409,46 @@ export class CoingeckoApi implements IPricesApi {
   /**
    * Requests the token prices provided by the CoinGecko API, using the {@link NetworkService}.
    */
-  private async _requestPricesFromNetwork(args: {
+  async _requestPricesFromNetwork(args: {
     chainName: string;
     tokenAddresses: Array<string>;
     fiatCode: string;
   }): Promise<Raw<AssetPrice>> {
+    const uniqueTokenAddresses = Array.from(
+      new Set(args.tokenAddresses.map((address) => address.toLowerCase())),
+    );
```

Comment:
Although not required, it makes sense to add this to reduce potential batch size.